linkerd/proxy/transport/src/listen.rs

```diff
@@ -64,10 +77,14 @@ where
         let server = Local(ServerAddr(listen.local_addr()?));
         let Keepalive(keepalive) = params.param();
         let accept = TcpListenerStream::new(listen).map(move |res| {
-            let tcp = res?;
+            let tcp = res.map_err(|e| io::Error::new(e.kind(), AcceptError(e)))?;
             super::set_nodelay_or_warn(&tcp);
-            let tcp = super::set_keepalive_or_warn(tcp, keepalive)?;
-            let client = Remote(ClientAddr(tcp.peer_addr()?));
+            let tcp = super::set_keepalive_or_warn(tcp, keepalive)
+                .map_err(|e| io::Error::new(e.kind(), KeepaliveError(e)))?;
+            let client = Remote(ClientAddr(
+                tcp.peer_addr()
+                    .map_err(|e| io::Error::new(e.kind(), PeerAddrError(e)))?,
```

Comment:
it seems a little weird that we wrap an `io::Error` in our various error types, and then wrap that error type in another `io::Error`? is there any reason that the `Bind` trait _needs_ to return an `io::Error` rather than a `linkerd_error::Error`? if we just changed the trait to return a `linkerd_error::Error`, we wouldn't need to wrap our errors in another layer of `io::Error`...

Reply:
There are assumptions in various places that the listeners/connectors return `io::Error` and I don't want to have to tackle all of that right now. I guess it would probably be fine to omit the `#[source]` annotation on these error types though.

Reply:
nevermind. fixed in https://github.com/linkerd/linkerd2-proxy/pull/1372/commits/f4b9d6f336e2990d778793ca5781cc027e99d663